zap/src/parser/grammar.lalrpop

```diff
@@ -103,6 +103,8 @@ TyKind: SyntaxTyKind<'input> = {
 
     "string" <r:("(" <IntRange> ")")?> => SyntaxTyKind::Str(r),
     "buffer" <r:("(" <IntRange> ")")?> => SyntaxTyKind::Buf(r),
+    "vector" "(" <x_ty:Ty> "," <y_ty:Ty> ")" => SyntaxTyKind::Vector(Box::new(x_ty), Box::new(y_ty), Box::new(SyntaxTy { start: 0, end: 0, kind: SyntaxTyKind::Num(NumTy::U8, None)})),
+    "vector" "(" <x_ty:Ty> "," <y_ty:Ty> "," <z_ty:Ty> ")" => SyntaxTyKind::Vector(Box::new(x_ty), Box::new(y_ty), Box::new(z_ty)),
```

Comment:
You should be able to call `vector` without specifying types and it would default to `f32, f32, f32`